### DISCUSSION TITLE
Code search unavailable when not logged in

### DISCUSSION THREAD

**Discussion by jaakristioja** [community/community#60237] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

Hello,

Due to reasons, I frequently have to use GitHub to browse the source code of multiple open-source repositories when not logged in. It seems that recently the feature of searching the source code of repositories has become unavailable for users not logged in, as the web interface redirect to a sign-in page in such cases.

I'm not sure whether or not this change was done due to performance or efficiency reasons.

Either way, code search not being available when not logged in has caused me to change my code search workflow. What I do now is (re)clone the respective repositories, `git grep` the source code, and likely `rm -rf` the cloned repositories if I don't plan to use the local copies later. This has resulted in me doing a lot of git cloning which requires a lot of time and network bandwidth, especially because most of the repositories are quite large and I have not yet memorized how to do shallow clones. I've also had to re-clone large repositories multiple times, having deleted them after individual searches.

I hope GitHub takes this into consideration, and again makes code search available for anonymous users, even if it is a simple grep-like search.

Thanks! :-)

**Comment by look** [comment, marked as answer]
Thanks for your feedback @jaakristioja. We now [require logging in to search code](https://github.blog/changelog/2023-06-07-code-search-now-requires-login/). Sorry this broke your workflow but we do not plan to change this at this time.
